### ISSUE TITLE
dh02_ce01 selects its template shape from a constant default, not from the model's L_IR (0.24 dex measured)

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2366]
`dh02_ce01` selects its template **shape** from a constant, not from the model's infrared luminosity.

The Dale & Helou (2002) / Chary & Elbaz (2001) library is tabulated on 161 `log10(L_TIR/L_sun)` nodes spanning 8.3–14.3, and the whole point of that family is that the SED shape correlates with luminosity — warmer, broader SEDs at higher L_IR. tengri's component interpolates the library correctly, but the value it interpolates at comes from a hardcoded closure default rather than from the galaxy being modelled.

`src/tengri/components/dust/emission/templates/dh02_ce01.py` (~line 1633):
```python
dust_log_lir: float = 10.0,   # constant; nothing derives it from L_ir
```
and the lookup a few lines below is otherwise sound:
```python
lir_c = jnp.clip(dust_log_lir, irlum_axis[0], irlum_axis[-1])
i = jnp.clip(jnp.searchsorted(irlum_axis, lir_c) - 1, 0, irlum_axis.shape[0] - 2)
f = (lir_c - irlum_axis[i]) / (irlum_axis[i + 1] - irlum_axis[i])
template_interp = (1.0 - f) * template_grid[i] + f * template_grid[i + 1]
```
The component's own module docstring already records the consequence: "The *shape* selected is always the log10(L_IR/L_sun) = 10 template, because the closure's `dust_log_lir` default is a constant and nothing derives it from `L_ir`."

**What this is NOT evidence of.** This was first noticed while chasing a 0.2406 dex shape residual in the agnfitter reproduction comparison, and that residual is *not* explained by this defect: pinning `dust_log_L_ir` to the realized 10.9505 so both sides read the same node leaves it at 0.2415 dex, unchanged. That comparison's residual remains an open question and is tracked with the page, not here. The defect below stands on the code and on the component's own docstring, independent of it.

**Who is affected.** Any `dust_emission={'type': 'dh02_ce01'}` model whose infrared luminosity is not ~1e10 L_sun gets the 1e10 shape with its own normalization — so the SED shape does not track luminosity, which is the property the library exists to provide. A fit is free to compensate by moving other parameters, so this is silent: the fit converges and the dust SED is the wrong temperature for its luminosity.

**Candidate fix.** Derive the default from the energy budget the component already receives: `dust_log_lir` should fall back to `log10(L_absorbed / L_sun)` (equivalently the `L_ir` the component publishes) when the user has not declared `dust_log_L_ir`, rather than to 10.0. `dust_log_L_ir` is already a declarable parameter that replaces energy balance when set, so the explicit spelling stays the override and only the default changes. That changes output for every existing `dh02_ce01` model, so it wants a regression test pinning the node selection against the realized luminosity, and a CHANGELOG note.

Two checks worth running with the fix: whether the same pattern exists in the sibling IR template components (a closure default that is not derived from the model's own published quantity is the general shape of this defect — `check_param_defaults.py` tests that defaults are inside their declared prior, not that they are derived), and whether any crossval test pins the current node-56 behaviour.
